src/components/App/index.js

```diff
@@ -1,39 +1,76 @@
-import React, {Component} from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { today } from 'constants/apiQueries';
+import { bindActionCreators } from 'redux';
 
+import { today } from 'constants/apiQueries';
 import { fetchEpisodes } from 'redux/actions/episodesActions';
 
+import ActiveEpisodeModal from 'components/App/ActiveEpisodeModal';
 import EpisodesList from 'components/App/EpisodesList';
 import Header from 'components/App/Header';
+import Pagination from 'components/App/Pagination';
 import Spinner from 'components/App/Spinner';
 
 import styles from './App.module.scss';
 
 class App extends Component {
+  state = {
+    isModalVisible: false,
+    activeEpisode: null,
+  }
+
   componentDidMount() {
-    this.props.dispatch(fetchEpisodes());
+    this.props.fetchEpisodes();
+  }
+
+  openModal = (episode) => {
+    this.setState({ 
+      isModalVisible: true,
+      activeEpisode: episode
+    });
+  }
+
+  closeModal = event => {
+    const { id } = event.target;
```

Comment:
you don't need any ide or event here, just set state of parent on click in child

Reply:
I didn't manage to handle this without id, it ended up with modal being closed on click anywhere. :/